### PR TITLE
`Development`: Update documentation for localVC SSH setup

### PR DESCRIPTION
**Pull request by SimonEntholzer**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

The documentation in the integrated code lifecycle setup is not up to date, and does not reference the SSH setup guide.

### Description

- Removed deprecated note
- Added link to SSH setup documentation
- Minor changes to SSH setup documentation 

### Steps for Testing

Clone the branch, and build the documentation. Check that everything is displayed as expected.

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

## Summary by CodeRabbit

- **Documentation**
	- Updated SSH configuration instructions to clarify steps for creating and managing SSH key pairs.
	- Emphasized correct SSH setup for cloning programming exercises in an integrated code lifecycle.
	- Adjusted file paths in SSH key generation commands for better accuracy.

### CODE REVIEW

File: docs/admin/setup/security.rst [hunk 1]
```diff
@@ -74,11 +79,12 @@ using the ssh-keygen command. Here's how you can do it:
 .. code-block:: bash
 
     # Generate RSA key pair
-    ssh-keygen -t rsa -b 4096 -f ~/.ssh/id_rsa
+    ssh-keygen -t rsa -b 4096 -f ~/id_rsa
 
     # Generate Ed25519 key pair
-    ssh-keygen -t ed25519 -f ~/.ssh/id_ed25519
+    ssh-keygen -t ed25519 -f ~/id_ed25519
```
[the remark this answers is not in the document]
Comment (SimonEntholzer) [reply docs/admin/setup/security.rst:85]: This change is deliberate, so the key creator does not overwrite their existing keys

**Review by b-fein — APPROVED** [review]

**Review by sarpsahinalp — DISMISSED** [review]
Lgtm I think creating a special subdirectory would also be a better solution

File: docs/admin/setup/security.rst [hunk 2]
```diff
@@ -74,11 +79,12 @@ using the ssh-keygen command. Here's how you can do it:
 .. code-block:: bash
 
     # Generate RSA key pair
-    ssh-keygen -t rsa -b 4096 -f ~/.ssh/id_rsa
+    ssh-keygen -t rsa -b 4096 -f ~/id_rsa
```
[the remark this answers is not in the document]
Comment (SimonEntholzer) [reply docs/admin/setup/security.rst:82]: good idea :+1:

File: docs/admin/setup/security.rst [hunk 3]
```diff
@@ -74,11 +79,12 @@ using the ssh-keygen command. Here's how you can do it:
 .. code-block:: bash
 
     # Generate RSA key pair
-    ssh-keygen -t rsa -b 4096 -f ~/.ssh/id_rsa
+    ssh-keygen -t rsa -b 4096 -f ~/artemis_ssh/id_rsa
```
[the remark this answers is not in the document]
Comment (SimonEntholzer) [reply docs/admin/setup/security.rst:82]: in fact this is not relevant here, as the permissions are changed afterwards anyways, in the place where the keys are moved to.

**Review by b-fein — APPROVED** [review]

**Review by coolchock — APPROVED** [review]

**Review by JohannesStoehr — APPROVED** [review]

**Review by florian-glombik — APPROVED** [review]
Looks fine, thanks for updating the docs 👍

File: docs/dev/setup/integrated-code-lifecycle.rst [hunk 4]
```diff
@@ -121,9 +121,7 @@ To create a course with registered users, you can use the scripts from ``support
     </iframe>
 
 - Make sure that the result of your submission is displayed in the Artemis UI.
-
-.. HINT::
-   At the moment, the Local VC system only supports accessing repositories via HTTP(S) and Basic Auth. We plan to add SSH support in the future. For now, you need to enter your Artemis credentials (username and password) when accessing template, solution, test, and assignment repositories.
+- Users can access their repositories via HTTPS and SSH. For SSH to work, you must first :ref:`configure SSH <configure-ssh-access>`.
```
Comment (florian-glombik) [inline docs/dev/setup/integrated-code-lifecycle.rst:124]: Can be found here https://artemis-platform--8835.org.readthedocs.build/en/8835/dev/setup/integrated-code-lifecycle.html#test-the-setup

**Pull request merged**